transformer_engine/pytorch/cpp_extensions/gemm.py

```diff
@@ -258,20 +276,44 @@ def gemm(
     fn = torch.ops.tex_ts.te_gemm_ts
     if ub_algo is not None:
         assert ub is not None, "ub object is None!"
-        if ub_algo == tex.UbufOverlapAlgo.BULK_OVERLAP_AG:
+        if ub_algo == tex.NVTE_Comm_Overlap_Algo.BULK_OVERLAP_AG:
+            if ub.is_fp8_ubuf():
+                if bulk_ubuf_fp8_type is not None:
+                    assert bulk_ubuf_fp8_type in [tex.DType.kFloat8E4M3, tex.DType.kFloat8E5M2]
+                else:
+                    bulk_fp8_dtype = tex.DType.kFloat8E4M3
+            else:
+                ubuf_dtype = ub.get_ubuf_output(tex.NVTE_Comm_Overlap_Type.AG).dtype
+                bulk_ubuf_fp8_type = TE_DType[ubuf_dtype]
             fn = ub.bulk_overlap
-            args = tuple(args + (1, empty_tensor))
-        elif ub_algo == tex.UbufOverlapAlgo.BULK_OVERLAP_RS:
+            args = tuple(
+                args
+                + (
+                    tex.NVTE_Comm_Overlap_Type.AG,
+                    bulk_ubuf_fp8_type,
+                    empty_tensor,
+                )
+            )
+        elif ub_algo == tex.NVTE_Comm_Overlap_Algo.BULK_OVERLAP_RS:
+            if ub.is_fp8_ubuf():
+                if bulk_ubuf_fp8_type is not None:
+                    assert bulk_ubuf_fp8_type in [tex.DType.kFloat8E4M3, tex.DType.kFloat8E5M2]
+                else:
+                    bulk_ubuf_fp8_type = tex.DType.kFloat8E4M3
+            else:
+                bulk_ubuf_fp8_type = TE_DType[
+                    ub.get_ubuf_output(tex.NVTE_Comm_Overlap_Type.AG).dtype
```

Comment:
Probably not a big deal if the sharded ("RS") and gathered ("AG") buffers have the same dtype, but asymmetry feels wrong to me:
```suggestion
                    ub.get_ubuf_output(tex.NVTE_Comm_Overlap_Type.RS).dtype
```

Reply:
We don't do anything with the buffer here except query the dtype, and requesting the AG (global) buffer avoids the compute to figure out the pointer offset for the calling rank. That may not be a meaningful optimization, but it's the reason for the asymmetry.